cmd/monitoring/monitoring.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/blackbox"
+	"github.com/openshift/openshift-azure/pkg/util/cloudprovider"
+	"github.com/openshift/openshift-azure/pkg/util/log"
+)
+
+var (
+	gitCommit = "unknown"
+	logLevel  = flag.String("loglevel", "Debug", "valid values are Debug, Info, Warning, Error")
+	interval  = flag.Duration("interval", 100*time.Millisecond, "check interval with dimension. Example: 1000ms ")
+	timeout   = flag.Duration("timeout", 2*time.Hour, "timeout when script will exit if left unattended")
```

Comment:
I think (and I would like to be proven differently) we need this because we start the process in the background during creation and it stays there during create, update, tests. So to make sure we dont leak because cleanup didn't ran we set a timeout on the overall process context

Reply:
with the right bash trap set up I think we can remove the timeout

Reply:
How I don't like traps :D